SCClassLibrary/Common/Streams/History.sc

```diff
@@ -5,6 +5,7 @@ History { 		// adc 2006, Birmingham; rewrite 2007.
 
 	classvar <>saveFolder = "~/Desktop/", <logFolder, <logFile, <logPath, <>keepsLog = true;
 	classvar <>current, <>maxShortLength=65;
+	classvar <>ignoreErrors = true;
```

Comment:
this is a code smell IMO. but if there's no way to just pass it directly to a method call, I guess it works.

Reply:
Generally it would be, yes :-) specifically here, it server two purposes:
when recording code experiments, you always write code that does not compile
or stops with errors, forgotten semicolons, etc etc. 
all these broken lines would clutter recorded history, so it is useful to have th option to just forget them.
when performing with history playback, such as (First Rule of the Avantgarde:) Try it Backwards. in that case, any lines that depend on historically previous ones will produce errors. 
```
a = [1, 2, 3]; // line 1
a.scramble; // line 2
// playing back line2 before line1 will produce 30 lines of Error post:
-> ERROR: Message 'scramble' not understood.
RECEIVER:
   nil
ARGS:
CALL STACK:
	DoesNotUnderstandError:reportError
		arg this = <instance of DoesNotUnderstandError>
	Nil:handleError
		arg this = nil
		arg error = <instance of DoesNotUnderstandError>
	Thread:handleError
		arg this = <instance of Thread>
		arg error = <instance of DoesNotUnderstandError>
	Object:throw
		arg this = <instance of DoesNotUnderstandError>
	Object:doesNotUnderstand
		arg this = nil
		arg selector = 'scramble'
		arg args = [*0]
	Interpreter:interpretPrintCmdLine
		arg this = <instance of Interpreter>
		var res = nil
		var func = <instance of Function>
		var code = "a.scramble"
		var doc = nil
		var ideClass = <instance of Meta_ScIDE>
	Process:interpretPrintCmdLine
		arg this = <instance of Main>
^^ The preceding error dump is for ERROR: Message 'scramble' not understood.
RECEIVER: nil
```
having your post window flooded with error messages in a show is annoying, and can slow down the IDE responses, so you want the option to suppress error posting - if you ever try turning it off, you will know exactly why you want it on ;-) 
makes sense?

Reply:
hm, not sure if that answers your comment - did it?

Reply:
my question was more about it having class scope as opposed to function scope, but I get what you're saying.

Reply:
The recording functions of `History` (and the related variables) are all class methods (class variables), because there is only one instance of `Interpreter` which we can add it to. I suppose instances could be useful for other kinds of logs, but this is not what the class is intended for currently.

The playback functions of `History` are all instance methods, because there are multple recordings that could be played back. So in such a way, it assumes that history happens only once, but leaves many traces.

I suspect that for clarity these could be two classes, or the recorder could have a single instance it redirects to. But for now, this looks good and has always worked rather well.